instrumentation/grape/lib/opentelemetry/instrumentation/grape/event_handler.rb

```diff
@@ -0,0 +1,125 @@
+      class EventHandler
+        class << self
+          # Handles the start of the endpoint_run.grape event (the parent event), where the context is attached
+          def endpoint_run_start(_name, _id, payload)
+            name = span_name(payload[:endpoint])
+            span = tracer.start_span(name, attributes: run_attributes(payload), kind: :server)
+            token = OpenTelemetry::Context.attach(OpenTelemetry::Trace.context_with_span(span))
+
+            payload.merge!(
+              __opentelemetry_span: span,
+              __opentelemetry_ctx_token: token
+            )
+          end
+
+          # Handles the end of the endpoint_run.grape event (the parent event), where the context is detached
+          def endpoint_run_finish(_name, _id, payload)
+            span = payload.delete(:__opentelemetry_span)
+            token = payload.delete(:__opentelemetry_ctx_token)
+            return unless span && token
+
+            if payload[:exception_object]
+              handle_error(span, payload[:exception_object])
+            else
+              span.set_attribute(OpenTelemetry::SemanticConventions::Trace::HTTP_STATUS_CODE, payload[:endpoint].status)
+            end
+
+            span.finish
+            OpenTelemetry::Context.detach(token)
+          end
+
+          # Handles the endpoint_render.grape event
+          def endpoint_render(_name, start, _finish, _id, payload)
+            name = span_name(payload[:endpoint])
+            attributes = {
+              'component' => 'template',
+              'operation' => 'endpoint_render'
+            }
+            tracer.in_span(name, attributes: attributes, start_timestamp: start, kind: :server) do |span|
+              handle_error(span, payload[:exception_object]) if payload[:exception_object]
+            end
+          end
+
+          # Handles the endpoint_run_filters.grape events
+          def endpoint_run_filters(_name, start, finish, _id, payload)
+            filters = payload[:filters]
+            type = payload[:type]
+
+            # Prevent submitting empty filters
+            zero_length = (finish - start).zero?
+            return if (!filters || filters.empty?) || !type || zero_length
+
+            name = span_name(payload[:endpoint])
+            attributes = {
+              'component' => 'web',
+              'operation' => 'endpoint_run_filters',
+              'grape.filter.type' => type.to_s
+            }
+            tracer.in_span(name, attributes: attributes, start_timestamp: start, kind: :server) do |span|
+              handle_error(span, payload[:exception_object]) if payload[:exception_object]
+            end
+          end
+
+          private
+
+          def tracer
+            Grape::Instrumentation.instance.tracer
+          end
+
+          def span_name(endpoint)
+            "#{api_instance(endpoint)} #{request_method(endpoint)} #{path(endpoint)}"
+          end
```

Comment:
I took the same approach to span names followed in the DDog instrumentation (see [here](https://github.com/DataDog/dd-trace-rb/blob/a02ccfc38123c087451f204ac6e822a6240216d5/lib/datadog/tracing/contrib/grape/endpoint.rb#L47)) and I'm following the [DDog porting guidelines](https://github.com/catawiki/opentelemetry-ruby-contrib/blob/55b6be133bd038fc597d8a08f7da661734eb0d38/instrumentation/datadog-porting-guide.md#L49), specifically the part where it says that `resource` in DDog is equivalent to `span.name` here.

I understand that we should keep cardinality low, but I'm also wondering how low is low? 🤔 And also, maybe a lower cardinality is needed for other instrumentations, but not for Grape? These are all open questions to me, and I would be really grateful if anybody could provide more input. I'll be happy to make the necessary adjustments to the span name.

Reply:
Yes, I totally agree. The reason I added it was because I was trying to follow the DDog porting guide, which said that resource -> span name, but I also think it's not necessary so I'm removing it 😄 
This way, we'll be following the recommended practice, as described [here](https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/http/#name).

Some background context on the low cardinality issue:

Span names are used to identify and group related traces. If span names have high cardinality , this can create performance issues. For example, if there are many different span names, it can be difficult to aggregate and analyse the data. In addition, high cardinality span names can cause excessive memory usage and slow down the tracing system.